lib/SILOptimizer/Differentiation/JVPCloner.cpp

```diff
@@ -315,19 +315,28 @@ class JVPCloner::Implementation final
   //--------------------------------------------------------------------------//
 
   void setTangentBuffer(SILBasicBlock *origBB, SILValue originalBuffer,
-                        SILValue tangentBuffer,
-                        llvm::Optional<IsInitialized_t> maybeInitInfo = None) {
+                        SILValue tangentBuffer, IsInitialized_t initInfo) {
     assert(originalBuffer->getType().isAddress());
     auto insertion =
         bufferMap.try_emplace({origBB, originalBuffer}, tangentBuffer);
     assert(insertion.second && "Tangent buffer already exists");
-    if (maybeInitInfo) {
-      auto initInfo = maybeInitInfo.getValue();
-      tangentBufferInitializationInfo[tangentBuffer] = initInfo;
-    }
+    assert(tangentBuffer->getType().isAddress());
+    tangentBufferInitializationInfo[tangentBuffer] = initInfo;
     (void)insertion;
   }
 
+  IsInitialized_t &getTangentBufferInitializationInfo(SILValue tangentBuffer) {
+    auto it = tangentBufferInitializationInfo.find(tangentBuffer);
```

Comment:
How about also adding an assertion in `getTangentBufferInitializationInfo`?
```cpp
    assert(tangentBuffer->getType().isAddress());
```

Reply:
It should already fail on `assert(it != .end())` since `assert(tangentBuffer->getType().isAddress())` is done on insertion to the map.